mobile-ui/src/main/java/com/playone/mobile/ui/firebase/v1/PlayoneFirebaseV1.kt

```diff
@@ -1,70 +1,192 @@
+    }
 
-    override fun kickNotification(payload: NotificationPayloadEntity) = TODO()
+    override fun getFavoritePlayoneList(
+        userId: Int,
+        callback: PlayoneCallback<List<PlayoneModel>>,
+        errorCallback: FirebaseErrorCallback
+    ) = favoriteDataSnapshot(userId.toString(), {}, errorCallback) {
+        favoriteSnap2PlayoneList(it, errorCallback, callback)
+    }
 
-    override fun quitNotification(payload: NotificationPayloadEntity) = TODO()
+    //region Fetching data from firebase database.
+    private fun <D> playoneDataSnapshot(
+        callback: PlayoneCallback<D>,
+        errorCallback: FirebaseErrorCallback,
+        strategy: DataSnapStrategy<D>
+    ) = dbReference
+        .child(GROUPS)
+        .addStrategyListener(callback, errorCallback, strategy)
+
+    private fun <D> playoneDataSnapshot(
+        id: String,
+        callback: PlayoneCallback<D>,
+        errorCallback: FirebaseErrorCallback,
+        strategy: DataSnapStrategy<D>
+    ) = dbReference
+        .child(GROUPS)
+        .child(id)
+        .addStrategyListener(callback, errorCallback, strategy)
+
+    private fun <D> userDataSnapshot(
+        userId: String,
+        callback: PlayoneCallback<D>,
+        errorCallback: FirebaseErrorCallback,
+        strategy: DataSnapStrategy<D>
+    ) = dbReference
+        .child(USERS)
+        .child(userId)
+        .addStrategyListener(callback, errorCallback, strategy)
+
+    private fun <D> joinedDataSnapshot(
+        userId: String,
+        callback: PlayoneCallback<D>,
+        errorCallback: FirebaseErrorCallback,
+        strategy: DataSnapStrategy<D>
+    ) = dbReference
+        .child(JOINED)
+        .child(userId)
+        .addStrategyListener(callback, errorCallback, strategy)
+
+    private fun <D> favoriteDataSnapshot(
+        userId: String,
+        callback: PlayoneCallback<D>,
+        errorCallback: FirebaseErrorCallback,
+        strategy: DataSnapStrategy<D>
+    ) = dbReference
+        .child(USERS)
+        .child(userId)
+        .child(FAVORITES)
+        .addStrategyListener(callback, errorCallback, strategy)
+    //endregion
+
+    //region Strategies of snapshot to the object.
+    private fun snap2PlayoneList(dataSnapshot: DataSnapshot?) =
+        dataSnapshot.takeIf(DataSnapshot?::isNotNull)
+            ?.children
+            ?.toMutableList()
+            ?.mapNotNull(::toPlayoneModel)
+            .orEmpty()
+
+    private fun snap2Playone(dataSnapshot: DataSnapshot?) =
+        dataSnapshot.takeIf(DataSnapshot?::isNotNull)?.run(::toPlayoneModel)
+
+    /**
+     * A strategy for transforming from user snapshot to a playone list thru a method
+     * [playoneDataSnapshot] to achieve it.
+     *
+     * @param dataSnapshot a snapshot from the firebase database.
+     * @param errorCallback a function for getting a error when retrieving the data.
+     * @param block a really block to get a [List].
+     */
+    private fun userSnap2PlayoneList(
```

Comment:
I this this is a valid naming according to https://android.github.io/kotlin-guides/style.html#naming-1
But I don't think in the name of `userSnap2PlayoneList` the `Snap2` is unusual constructs.
> Sometimes there is more than one reasonable way to convert an English phrase into camel case, such as when acronyms or unusual constructs like “IPv6” or “iOS” are present. To improve predictability, use the following scheme.

I know what you name this function but my opinion is using `userSnapToPlayoneList` is better.